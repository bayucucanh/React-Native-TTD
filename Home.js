import { StyleSheet, Text, View } from 'react-native'
import React from 'react'
import Profile from './Profile'

class Home extends React.Component {
  constructor() {
    super()
    this.state = {
      data: ''
    }
  }

  change(x) {
    this.setState({ data: x * 10})
  }

  tambah(x) {
    this.setState({ data: x + 10})
  }

  kurang(x) {
    this.setState({ data: x - 10})
  }

  bagi(x) {
    this.setState({ data: x / 2})
  }

  render() {
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <Profile />
      <Text style={{ fontSize: 10 }}>Home Component</Text>
    </View>
  )
}
}

export default Home

const styles = StyleSheet.create({})