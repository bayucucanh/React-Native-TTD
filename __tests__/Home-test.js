import 'react-native'
import React from 'react'
import Home from '../Home'
import renderer from 'react-test-renderer'

test('Home snapshot', () => { 
  const snap = renderer.create(
    <Home />
  ).toJSON();
  expect(snap).toMatchSnapshot()
 })