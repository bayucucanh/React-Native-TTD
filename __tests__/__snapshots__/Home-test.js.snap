// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text>
      Profile
    </Text>
  </View>
  <Text
    style={
      Object {
        "fontSize": 10,
      }
    }
  >
    Home Component
  </Text>
</View>
`;
